Python/Product/Analysis/ModuleResolver.cs

```diff
@@ -0,0 +1,206 @@
+    internal class ModuleResolver {
+        /// <summary>
+        /// Resolves modules listed in the relative import statetement
+        /// such as 'from . import a, b, c' or 'from .a import b, c'
+        /// </summary>
+        /// <param name="entry">Project entry</param>
+        /// <param name="node">Import statement node</param>
+        /// <returns>Names of modules</returns>
+        internal static IEnumerable<string> ResolveRelativeFromImport(IPythonProjectEntry entry, FromImportStatement node)
+            => ResolveRelativeFromImport(entry.ModuleName, entry.FilePath, node);
+
+        /// <summary>
+        /// Resolves modules listed in the relative import statetement
+        /// such as 'from . import a, b, c' or 'from .a import b, c'
+        /// </summary>
+        /// <param name="importingFromModuleName">Name of the importing module</param>
+        /// <param name="importingFromFilePath">Disk path to the importing module</param>
+        /// <param name="node">Import statement node</param>
+        /// <returns></returns>
+        internal static IEnumerable<string> ResolveRelativeFromImport(string importingFromModuleName, string importingFromFilePath, FromImportStatement node) {
+            var root = node.Root.MakeString();
+
+            if (!string.IsNullOrEmpty(root) && root.StartsWith(".")) {
+                var prefix = root.All(c => c == '.') ? root : $"{root}.";
+
+                var names = node.Names.Where(n => !string.IsNullOrEmpty(n.Name)).Select(n => n.Name);
+                if (!names.Any()) {
+                    return Enumerable.Empty<string>();
+                }
+
+                var resolved =  names.SelectMany(n => ResolvePotentialModuleNames(importingFromModuleName, importingFromFilePath, $"{prefix}{n}", node.ForceAbsolute)).ToArray();
+                if (resolved.Length == 1 && resolved[0].Length > 2 && resolved[0].EndsWithOrdinal(".*")) {
+                    resolved[0] = resolved[0].Substring(0, resolved[0].Length-2);
+                }
+                return resolved;
+            }
+
+           return ResolvePotentialModuleNames(importingFromModuleName, importingFromFilePath, root, node.ForceAbsolute);
+        }
+
+        /// <summary>
+        /// Extracts actual module names from the 'from import' statetement
+        /// such as [a, b, c] from 'from . import a, b, c' or [a] from 'from .a import b, c'
+        /// </summary>
+        /// <param name="entry">Project entry</param>
+        /// <param name="node">Import statement node</param>
+        /// <returns>Names of modules</returns>
+        internal static IEnumerable<string> GetModuleNamesFromImport(IPythonProjectEntry entry, FromImportStatement node)
+            => GetModuleNamesFromImport(entry.ModuleName, entry.FilePath, node);
+
+        /// <summary>
+        /// Extracts actual module names from the 'from import' statetement
+        /// such as [a, b, c] from 'from . import a, b, c' or [a] from 'from .a import b, c'
```

Comment:
These seem like inconsistent results. Given the first example, I'd expect `a.b, a.c` from the second, and it's not clear how this relates to the module it's being imported from (really, I'd expect `importingFromModuleName.a, ...` in the results)